### ISSUE TITLE
A false negative about the rule NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE

### ISSUE THREAD

**Issue by LynnBroe** [spotbugs/spotbugs#2277]
Version: 4.7.3

I found a false negative about the rule [NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#np-parameter-must-be-non-null-but-is-marked-as-nullable-np-parameter-must-be-nonnull-but-marked-as-nullable). In the example below, SpotBugs should have reported a warning at line 6 because the value of `foo` is from a method annotated with `Nullable`. Besides, this annotation is from SpotBugs. Hence, I think this is a false negative.

```java
import edu.umd.cs.findbugs.annotations.Nullable;

public class C {
   public static int h() {
       final C foo = C.something();
       return foo.hashCode();  // should report a warning at this line
   }

   @Nullable
   private static C something() {
       return null;
   }
}
```
